src/Control/Monad/Free.hs

```diff
@@ -11,6 +11,23 @@
 #ifndef MIN_VERSION_base
 #define MIN_VERSION_base(x,y,z) 1
 #endif
+
+#if MIN_VERSION_base(4,9,0)
+#define LIFTED_FUNCTOR_CLASSES 1
+#else
+#if MIN_VERSION_transformers(0,5,0)
+#define LIFTED_FUNCTOR_CLASSES 1
+#else
+#if MIN_VERSION_transformers_compat(0,5,0) && !MIN_VERSION_transformers(0,4,0)
+#define LIFTED_FUNCTOR_CLASSES 1
+#endif
+#endif
+#endif
```

Comment:
I'm not terribly fond of this chunk of CPP. I'd vastly prefer detecting whether we're using a version of `base`/`transformers`/`transformers-compat` that supports the lifted classes in the `.cabal` file itself (I [do this](https://github.com/RyanGlScott/text-show/blob/8e92bd9f45f156e4219037e7ff886527935d96d1/text-show.cabal#L181-L198) in `text-show`, for instance).

Also, when we adapt this approach to other modules, we won't have to copy/paste this whole thing in a bunch of other places.